### PR TITLE
Throw a Not Found error

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Currently an unmatched route ends up with the status code being `404` rather than an error thrown. This converts any error status code (where no body has been set) to a `HttpError` and throws it. This will allow all errors to be caught consistently, but has no visible effect right now.

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
(Feature or refactor, given there's no visible change? 🤔)

**Comment by diversemix** [comment]
Could you give a few more words why this is being done?

**Comment by thewilkybarkid** [comment]
As part of https://github.com/libero/publisher/issues/314 we'll be making sure error responses are in the Hydra format. Currently 404s aren't thrown errors, so need to handle both thrown errors and recognise a response with a 404 status code and no body.

**Comment by thewilkybarkid** [comment]
Other option could be to make `@koa/router` throw a 404 if no routes match...

**Review by diversemix — APPROVED** [review]

File: src/middleware/http-errors.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Context, Middleware, Next } from 'koa';
```
Comment (giorgiosironi) [inline src/middleware/http-errors.ts]: `middleware/http-errors` has the same base name as the library `http-errors`, possible confusion when looking at the imported module in the middle of other files. Is it the most fitting name?

**Comment by thewilkybarkid** [comment]
I'm going to simplify this, in favour of getting the router to throw a `NotFound` error on a 404.

File: src/middleware/routing.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    if (typeof _matchedRoute === 'undefined') {
+      throw new createHttpError.NotFound();
+    }
+  }
+);
+
+export default (router: Router): Middleware => (
+  compose([
+    router.routes(),
+    notFound(),
```
Comment (thewilkybarkid) [inline src/middleware/routing.ts:19]: Will propose changing the library to allow `router.routes({ throw: true })`.

**Review by giorgiosironi — APPROVED** [review]

**Pull request merged**